src/osd/PrimaryLogPG.cc

```diff
@@ -141,6 +141,10 @@ class PrimaryLogPG::BlessedGenContext : public GenContext<T> {
       c.release()->complete(t);
     pg->unlock();
   }
+  bool sync_finish(T t) {
+    c.release()->complete(t);
```

Comment:
Shouldn't it call (and return result of) sync_complete instead? The same below...